packages/@lwc/ssr-compiler/src/compile-js/index.ts

```diff
@@ -111,14 +107,29 @@ const visitors: Visitors = {
             return;
         }
 
-        const decorators = (node as DecoratatedMethodDefinition).decorators;
-        const decoratedExpression = decorators?.[0]?.expression as Expression;
+        const { decorators } = node;
+        // The real type is a subset of `Expression`, which doesn't work with the `is` validators
+        const decoratedExpression = decorators?.[0]?.expression;
         if (
             is.callExpression(decoratedExpression) &&
             is.identifier(decoratedExpression.callee) &&
             decoratedExpression.callee.name === 'wire'
         ) {
-            catalogWireAdapters(path, state);
+            // Getters and setters are methods in the AST, but treated as properties by @wire
+            // Note that this means that their implementations are ignored!
+            if (node.kind === 'get' || node.kind === 'set') {
+                const methodAsProp = b.propertyDefinition(
+                    node.key,
+                    b.identifier('undefined'),
+                    node.computed,
+                    node.static
+                );
+                methodAsProp.decorators = decorators;
+                path.replaceWith(methodAsProp);
+                return;
```

Comment:
It looks like we're missing a `catalogWireAdapters()` here. Presumably a test would fail if we needed to validate the wire ID/config?

Reply:
We don't need to catalog getter/setters, because they're not going to exist. The cataloging gets done by the `PropertyDefinition` visitor when we visit the newly-created node.